### PR TITLE
jasmine trim tests using describe and it.

### PR DESCRIPTION
**Pull request by ocampd1**
Added more unit tests to trim-method.

### CODE REVIEW

File: src/main/javascript/trim.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
```
Comment (tonte-pouncil) [inline src/main/javascript/trim.html]: Remove this file from project and create your on js-tutorial project and then add it to that project.  I can help you with this if you need me too.  Just let me know.

File: src/test/javascript/stringExtensionSpec.js [hunk 2]
```diff
@@ -1,46 +1,71 @@
-define(['bytepushers'], function(BytePushers) {
-    describe("String extension tests:", function() {
+define(['bytepushers'], function (BytePushers) {
+    describe("String extension tests:", function () {
 
         var testStr = "Hello this is a random string";
 
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
+            it('will trim whitespace correctly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 3]
```diff
@@ -1,46 +1,71 @@
-define(['bytepushers'], function(BytePushers) {
-    describe("String extension tests:", function() {
+define(['bytepushers'], function (BytePushers) {
+    describe("String extension tests:", function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:2]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 4]
```diff
@@ -1,46 +1,71 @@
-define(['bytepushers'], function(BytePushers) {
-    describe("String extension tests:", function() {
+define(['bytepushers'], function (BytePushers) {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 5]
```diff
@@ -1,46 +1,71 @@
...
         var testStr = "Hello this is a random string";
 
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
+            it('will trim whitespace correctly', function () {
                 var result = "                       l     shdasdk;lg;al j          ".trim();
 
                 expect(result).toBe("l     shdasdk;lg;al j");
             });
+            it("Will Not Trim in Front of a String", function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 6]
```diff
@@ -1,46 +1,71 @@
...
+                var frontTrimTest = "ABC";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename frontTrimTest variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js [hunk 7]
```diff
@@ -1,46 +1,71 @@
...
+                var expectedResult = " ABC";
+                var actualResult = frontTrimTest.trim();
+
+                expect(expectedResult).not.toBe(actualResult);
+                //expect(false).not.toBe(true);
+            });
+
+            it("Will Trim in Front of a String", function () {
+                var frontTrimTest = " ABC";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename frontTrimTest variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js [hunk 8]
```diff
@@ -1,46 +1,71 @@
...
+                var expectedResult = "ABC";
+                var actualResult = frontTrimTest.trim();
+
+                expect(expectedResult).toBe(actualResult);
+                //expect(false).not.toBe(true);
+            });
+
+            it("Will Not Trim", function () {
+                var noSpaceTrimTest = " ABC ";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename noSpaceTrimTest variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js [hunk 9]
```diff
@@ -1,46 +1,71 @@
...
+                var expectedResult = "ABC";
+                var actualResult = noSpaceTrimTest.trim();
+                expect(actualResult).toBe(expectedResult);
+
+            });
         });
 
         describe('String.prototype.includes', function () {
-            it('can tell if a string includes a string', function() {
+            it('can tell if a string includes a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 10]
```diff
@@ -1,46 +1,71 @@
...
                 var result = testStr.includes("o this i");
 
                 expect(result).toBe(true);
             });
 
-            it('can tell if a string does not include a string', function() {
+            it('can tell if a string does not include a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 11]
```diff
@@ -1,46 +1,71 @@
...
                 var result = testStr.includes("This is not in the string");
 
                 expect(result).toBe(false);
             });
         });
 
         describe('String.prototype.toCamelCase', function () {
-            it('can format a sentence into camel case', function() {
+            it('can format a sentence into camel case', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 12]
```diff
@@ -1,46 +1,71 @@
...
                 var result = "Hello this is a sentence.".toCamelCase();
 
                 expect(result).toBe("helloThisIsASentence.");
             });
         });
 
         describe('String.prototype.toNormalCase', function () {
-            it('can format a sentence into camel case', function() {
+            it('can format a sentence into camel case', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 13]
```diff
@@ -1,46 +1,71 @@
...
                 var result = "Hello world".toCamelCase().toNormalCase();
                 // helloWorld => "Hello World"
                 expect(result).toBe("Hello World");
             });
 
-            it('can format a sentence into camel case with a single letter word', function() {
+            it('can format a sentence into camel case with a single letter word', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 14]
```diff
@@ -51,7 +76,7 @@ define(['bytepushers'], function(BytePushers) {
         });
 
         describe('String.format', function () {
-            it('can format a string properly', function() {
+            it('can format a string properly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

**Review by tonte-pouncil — CHANGES_REQUESTED** [review]

File: src/test/javascript/stringExtensionSpec.js (hunk 2 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 3 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js:2]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 4 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parentheses.

File: src/main/javascript/trim.html (hunk 1 above)
Comment (ocampd1) [reply src/main/javascript/trim.html]: Moved to new project file.

File: src/test/javascript/stringExtensionSpec.js (hunk 5 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parenthesis.

File: src/test/javascript/stringExtensionSpec.js (hunk 10 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parenthesis.

File: src/test/javascript/stringExtensionSpec.js (hunk 11 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parenthesis.

File: src/test/javascript/stringExtensionSpec.js (hunk 13 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parenthesis.

File: src/test/javascript/stringExtensionSpec.js (hunk 14 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parenthesis.

File: src/test/javascript/stringExtensionSpec.js (hunk 12 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parenthesis.

File: src/test/javascript/stringExtensionSpec.js (hunk 6 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Renamed from frontTrimTest variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js (hunk 8 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Renamed from noSpaceTrimTest variable to stringUnderTest.

File: src/test/javascript/stringExtensionSpec.js (hunk 9 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Removed space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js (hunk 7 above)
Comment (ocampd1) [reply src/test/javascript/stringExtensionSpec.js]: Renamed from frontTrimTest variable to stringUnderTest.

**Review by tonte-pouncil — APPROVED** [review]
Nice Job!

**Pull request merged**